src/activities/quizzes/timing/QuizTimingEntity.js

```diff
@@ -79,13 +85,58 @@ export class QuizTimingEntity extends Entity {
 		return entity.properties && entity.properties.timeLimit;
 	}
 
-	async setTimingType(data) {
+	async setExtendedDeadline(data) {
 		if (!this.canEditTiming()) return;
+		const entity = this.getEnforcedTimingSubEntity();
+		if (!entity) return;
+		const subEntity = entity.getSubEntityByClass('automatic-zero');
+		const action = subEntity.getActionByName(Actions.quizzes.timing.updateTimingLateData);
+		const fields = [
+			{ name: 'submissionLateData', value: data }
+		];
+		await performSirenAction(this._token, action, fields);
+	}
+
+	async setGracePeriod(data) {
+		if (!this.canEditTiming()) return;
+		const entity = this.getEnforcedTimingSubEntity();
+		if (!entity) return;
+		const action = entity.getActionByName(Actions.quizzes.timing.updateTimingGraceLimit);
```

Comment:
For this and the other set functions, we should first check whether the action we want exists using `entity.hasActionByName` instead of using `this.canEditTiming()`.

Reply:
sounds good, I made 3 new helpers to check the actions